rts/motoko-rts/src/persistence/compatibility.rs

```diff
@@ -110,16 +110,16 @@ unsafe fn create_type_check_cache<M: Memory>(
     old_type: &TypeDescriptor,
     new_type: &TypeDescriptor,
 ) -> BitRel {
-    let old_type_count = old_type.type_count() as u32;
-    let new_type_count = new_type.type_count() as u32;
+    let old_type_count = old_type.type_count();
+    let new_type_count = new_type.type_count();
```

Comment:
`u32`s would suffice for the counts, I think.

Reply:
I agree. I only upgraded this for symmetry and simplicity. 
I'll put it on the backlog for `u32` downgrade.